src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColService.java

```diff
@@ -366,20 +368,22 @@ public boolean check_A_And_B_Same_Order(List<String> elementsA, List<String> ele
 
     /**
      * Construct a seqCol level 2 (Map representation) out of the given seqColL2Map*/
-    public Map<String, String> constructSeqColLevelOneMap(Map<String, List<String>> seqColL2Map) throws IOException {
+    public Map<String, String> constructSeqColLevelOneMap(Map<String, List<?>> seqColL2Map) throws IOException {
         Map<String, String> seqColL1Map = new TreeMap<>();
         Set<String> seqColAttributes = seqColL2Map.keySet(); // The set of the seqCol attributes ("lengths", "sequences", etc.)
         for (String attribute: seqColAttributes) {
             String attributeDigest = digestCalculator.getSha512Digest(
-                    convertSeqColLevelTwoAttributeValuesToString(seqColL2Map.get(attribute)));
+                    convertSeqColLevelTwoAttributeValuesToString(seqColL2Map.get(attribute),
+                                                                 SeqColExtendedDataEntity.AttributeType.valueOf(
+                                                                         attribute)));
```

Comment:
Would this throw an exception if this seq col map (which comes from the requester, if I understand correctly) had an attribute we didn't support?

Reply:
I feel like for attributes not supported in our implementation, we can't always know the correct `toString`, and I think that should be okay - we're not going to be able to compare the values in unsupported attributes, all we can do is say that those attributes exist in the user-provided collection when we return the comparison result.

For example say we have this seqcol in our db:
```
{
    "names": ["A", "B", "C"],
    "lengths": [11, 22, 33]
}
```

And the user wants to compare it to this one:
```
{
    "names": ["A", "B", "C"],
    "lengths": [11, 22, 33],
    "myUnsupportedStrAttribute": ["1", "2", "3"],
    "myUnsupportedIntAttribute": [1, 2, 3]
}
```

First of all, I realised I don't actually know how Spring figures out what types to fill in for the request body [here](https://github.com/EBIvariation/eva-seqcol/blob/7deaef341eef87890678ad6935c9f66eeca8bb91/src/main/java/uk/ac/ebi/eva/evaseqcol/controller/seqcol/SeqColComparisonController.java#L88)... if you've tried it out I'd be curious to know!

But regardless, I don't think we're going to do anything with those arrays or their digests when we do the comparison, the attribute names will just show up in the `b_only` part of the comparison result.

Reply:
Anyway, I agree it can be addressed/discussed later :+1:

Reply:
@apriltuesday 
> First of all, I realised I don't actually know how Spring figures out what types to fill in for the request body [here](https://github.com/EBIvariation/eva-seqcol/blob/7deaef341eef87890678ad6935c9f66eeca8bb91/src/main/java/uk/ac/ebi/eva/evaseqcol/controller/seqcol/SeqColComparisonController.java#L88)... if you've tried it out I'd be curious to know!

That's a very good point, I wasn't completely aware of how it worked myself. 

However after running some tests, I figured out that Spring can figure out the type of the inputted values, one by one. 
For example, I posted a seqcol object with the following `lengths` array:
```json
{ ...
"lengths" : ["11111","22222", 230218, 813184, ...]
...}
```
And tried to print out the type of each attribute, and I got the following:
```shell
lengths[0] of type java.lang.String
lengths[1] of type java.lang.String
lengths[2] of type java.lang.Integer
lengths[3] of type java.lang.Integer
```
And since java `Lists` can contain multiple types, we had no issue.

So thanks April for pointing that out, and we should be definitely cautious when using this. 

Because a user can provide a valid seqcol attribute's name such as `lengths`, and give values `["11111","22222", 230218, 813184, ...]` and we won't be able to detect that.
Normally, "11111" and 11111 will be treated as non-equal in the comparison function and won't have a problem, however I still think that we should make a control on what the user should POST.